crypten/communicator/distributed_communicator.py

```diff
@@ -264,7 +264,7 @@ def send_obj(self, obj, dst, group=None):
 
         buf = pickle.dumps(obj)
         size = torch.tensor(len(buf), dtype=torch.int32)
-        arr = torch.from_numpy(numpy.frombuffer(buf, dtype=numpy.int8))
+        arr = torch.from_numpy(numpy.copy(numpy.frombuffer(buf, dtype=numpy.int8)))
```

Comment:
From what I have read from here[1], the ```arr``` should share the same memory with the numpy array in case ```numpy.copy``` is not used.

Q: It is needed to create a copy of the data? (by using this I think we are playing it safe, but if ```buf``` becomes bigger it might introduce an overhead because of the copy)
 
[1] https://pytorch.org/docs/stable/generated/torch.from_numpy.html

Reply:
Without this copy, I am seeing [this warning](https://github.com/pytorch/pytorch/blob/master/torch/csrc/utils/tensor_numpy.cpp#L135-L141). I tried altering the writeable flag of the numpy array directly but that does not appear to be allowed here. The copy makes the warning go away.

Reply:
Yep, you are right!

I have read [here](https://github.com/pytorch/pytorch/issues/37581) that pytorch does not support for the moment read-only tensors!